duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/helper/DuckChatJSHelper.kt

```diff
@@ -183,6 +201,82 @@ class RealDuckChatJSHelper @Inject constructor(
         }
     }
 
+    /**
+     * Accept incoming JSON payload { "serializedMigrationFile": "..." }
+     * Store the string value in an ordered list for later retrieval
+     */
+    private fun getStoreMigrationDataResponse(
+        featureName: String,
+        method: String,
+        id: String,
+        data: JSONObject?,
+    ): JsCallbackData {
+        val item = data?.optString(SERIALIZED_MIGRATION_FILE)
+        val jsonPayload = JSONObject()
+        if (item != null && item != JSONObject.NULL) {
+            synchronized(migrationItems) { migrationItems.add(item) }
```

Comment:
why do you need a synchronized here? If you are going to block the main thread it should not be here and moved to a background thread.